lib/double_entry/transfer.rb

```diff
@@ -72,34 +71,52 @@ def initialize(attributes)
       end
     end
 
-    def process(amount, from, to, code, detail)
-      if from.scope_identity == to.scope_identity && from.identifier == to.identifier
-        fail TransferNotAllowed, 'from and to are identical'
+    def process(amount, options)
+      from_acct = options[:from]
+      to_acct = options[:to]
+      code = options[:code]
+      detail = options[:detail]
+      metadata = options[:metadata]
+      if from_acct.scope_identity == to_acct.scope_identity && from_acct.identifier == to_acct.identifier
+        fail TransferNotAllowed, 'from_acct and to_acct are identical'
```

Comment:
if they are still to remain as `from` and `to` in the argument list, perhaps the error message should conform to that?